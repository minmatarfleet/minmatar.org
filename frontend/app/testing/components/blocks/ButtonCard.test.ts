import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import ButtonCard from "@components/blocks/ButtonCard.astro";

test("ButtonCard defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(ButtonCard, {
    props: {
      title: "chickens rule the world",
      href: "chickens.com",
    },
  });

  expect(result).toMatchSnapshot();
});
