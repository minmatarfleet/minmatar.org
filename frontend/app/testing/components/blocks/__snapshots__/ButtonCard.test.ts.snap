// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonCard defaults 1`] = `"<a class="button-card" href="chickens.com" data-astro-cid-x44kssfb> <div class="wrapper [ card component ] light-transparency" data-astro-cid-x44kssfb="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: 0;--wrapper_padding_inline_end: 0;--wrapper_padding_block: var(--space-l);--wrapper_max_width: 100%;">   <div class="flex-block items-center gap---space-3xs" data-astro-cid-x44kssfb="true" data-mod style="--flex_block_gap: var(--space-3xs);">  <picture data-astro-cid-x44kssfb>  </picture> <div class="flex-block [ text-center ] gap---space-3xs" data-astro-cid-x44kssfb="true" data-mod style="--flex_block_gap: var(--space-3xs);">  <h3 data-astro-cid-x44kssfb>chickens rule the world</h3>  </div>   </div>    </div>   </a> "`;
